pkg/sql/truncate.go

```diff
@@ -131,6 +131,8 @@ func (p *planner) truncateTable(ctx context.Context, id sqlbase.ID, traceKV bool
 		return err
 	}
 	newTableDesc := *tableDesc
+	newTableDesc.ReplacementOf = append(newTableDesc.ReplacementOf,
+		sqlbase.TableDescriptor_Replacement{ID: id, Time: p.txn.OrigTimestamp()})
```

Comment:
Is there anyway we can bound this? I know we can use the zonecfg read below and use zoneCfg.GC.TTLSeconds to check if any old entries have expired, but that still doesn't prevent 
thousands of TRUNCATE operations on the same table to append entries here. Would backup be
okay with a limit of 10 or something.

Reply:
My original revision just stored the ID of the immediate predecessor table that this one replaced, rather than the repeated field, effectively forming a linked list from descriptor to descriptor that would, in-effect, be self-pruning, as the old descriptors are GC'ed.

BACKUP needs all the revisions in the time window, which is a subset of everything in the GC window, so it would be fine with those semantics. If we use a repeated field and do our own pruning, we'd need to make sure it results in the same (or greater) retention, so a const like 10 probably isn't OK, but something based on TTL would be.

Reply:
It is not self pruning because the list is on the latest descriptor. I think you could decode the zonecfg you're reading later on in this function and pick up the TTL from there to prune this list.

Reply:
er, sorry, I meant it was self-pruning when each descriptor pointed only to its immediate predecessor, with the optional field, rather than keeping a list in a repeated field.

That said, it would make it a bit harder to use in BACKUP though, though since we'd need to do a historical read of all descriptors to find those previous versions to walk, even for non-`revision_history` backups, so maybe the list on the descriptor is a tad easier to use even if we need to add our own pruning.